eZ/Publish/Core/Repository/UserService.php

```diff
@@ -781,52 +762,12 @@ public function updateUserPassword(APIUser $user, string $newPassword): APIUser
             throw new MissingUserFieldTypeException($loadedUser->getContentType(), self::USER_FIELD_TYPE_NAME);
         }
 
-        $contentUpdateStruct = $contentService->newContentUpdateStruct();
-        $contentUpdateStruct->setField(
-            $userFieldDefinition->identifier,
-            new UserValue([
-                'contentId' => $loadedUser->id,
-                'hasStoredLogin' => true,
-                'login' => $loadedUser->login,
-                'email' => $loadedUser->email,
-                'plainPassword' => $newPassword,
-                'enabled' => $loadedUser->enabled,
-                'maxLogin' => $loadedUser->maxLogin,
-                'passwordHashType' => $user->hashAlgorithm,
-                'passwordHash' => $user->passwordHash,
-            ])
-        );
-
-        $errors = $this->contentValidator->validate(
-            $contentUpdateStruct,
-            ['content' => $loadedUser],
-        );
-
+        $errors = $this->passwordValidator->validatePassword($newPassword, $userFieldDefinition);
```

Comment:
`$errors` returned here are instances of `eZ\Publish\SPI\FieldType\ValidationError` instead of `eZ\Publish\Core\FieldType\ValidationError`, but this validator is used internally in the User Type and I don't see any other exception class that would fit. WDYT?

Reply:
> `$errors` returned here are instances of `eZ\Publish\SPI\FieldType\ValidationError` instead of `eZ\Publish\Core\FieldType\ValidationError`, but this validator is used internally in the User Type and I don't see any other exception class that would fit. WDYT?

If it's already provided like that, then ok, we can re-use it. Ideally there should be a separate set of exceptions for this, but maybe an overkill...

Reply:
@alongosz 
From what I've seen Field Types create Core Validation Errors, even though they specify that they would return arrays of SPI Validation Errors (which are not translatable). 
Exception class itself defines Core Validation Errors as acceptable (which excludes their parent SPI variant), but does not complain / does any checks against types. `DomainContentMutationResolver` that is one class that I've found that uses this exception does relies on specific array structure inside the exception (namely, by language codes first) and needs them to be translatable.

What approach do you imagine would be best?